src/main/java/ulcrs/controllers/CourseController.java

```diff
@@ -19,11 +17,7 @@
     public RouteGroup routes() {
         return () -> {
             before("/*", (request, response) -> log.info("endpoint: " + request.pathInfo()));
-            get("/", this::getCourseList, courses -> {
-            	// Return only the required fields in JSON response
-            	Gson gson = new GsonBuilder().excludeFieldsWithoutExposeAnnotation().setPrettyPrinting().create();
-            	return gson.toJson(courses);	
-            });
+            get("/", this::getCourseList, gson::toJson);

```

Comment:
The GsonFactory approach works or this `/` endpoint, though we don't want to exclude @Expose annotations always, which I believe will happen when using this Factory. For instance, for the `/{id}` endpoint we want all attributes, not just the attributes labelled with @Expose. I believe the gson::toJson call will return only attributes labelled with @Expose, which we don't want for the `/{id}` endpoint. You'll want to confirm this is or isn't the case, though from my testing, I believe it may be the case.

Reply:
You are right. Changed to use exposeOnlyGson in factory